src/ServiceProvider.php

```diff
@@ -14,8 +12,8 @@ final class ServiceProvider extends AbstractServiceProvider
      */
     public function register()
     {
-        $this->app->extend(Engine::class, static function (Engine $engine, Application $app) {
-            return $app->make(EngineDecorator::class, compact('engine'));
+        $this->app->bind(\ElasticScoutDriver\Engine::class, static function (Application $app) {
```

Comment:
This looks a bit hacky cause we replace one class with another, it would be better to do:

```php
resolve(EngineManager::class)->extend('elastic', static function () {
            return resolve(\ElasticScoutDriverPlus\Engine::class);
});
```

But we can't guarantee, which package will be loaded first, so this will not work. Let's maybe still use `extend` instead of `bind`, because it fits semantically better. Also, please use short class names (i.e.  `Engine` and `EnginePlus` alias).